metricbeat/helper/module.go

```diff
@@ -144,26 +152,18 @@ func (m *Module) Run(period time.Duration, b *beat.Beat) {
 	go m.publishing(b)
 
 	// Start fetching metrics
+	// The frequency is always based on the ticker interval. No delays are taken into account.
+	// Each Metricset must ensure to exit after timeout.
```

Comment:
Like @andrewkroh was suggesting, could we keep a counter with the number of goroutines started but not exit? We can then stop creating new ones if the previous 60 or so didn't exit yet, and log an error instead. This is to protect against "too many open files" and excessive memory usage. Good to have this in a follow up PR if you prefer.